drivers/iio/proximity/vcnl3020.c

```diff
@@ -0,0 +1,338 @@
+#include <linux/iio/sysfs.h>
+#include <linux/platform_device.h>
+#include <linux/iio/proximity/vcnl3020.h>
+
+static const struct i2c_device_id vcnl3020_id[] = {
+	{ "vcnl3020", 0 },
+	{}
+};
+MODULE_DEVICE_TABLE(i2c, vcnl3020_id);
+
+static int32_t vcnl3020_init(struct vcnl3020_data *data)
+{
+	s32 rc;
+	u32 proximity_rate, led_current, threshold, count_exceed, param_val;
+
+	struct device *dev = &data->client->dev;
+
+	rc = device_property_read_u32(dev, "proximity-rate", &param_val);
+	if (rc) {
+		dev_err(dev, "couldn't get proximity rate %x", rc);
+		goto exit;
+	}
+	proximity_rate = param_val;
+
+	rc = device_property_read_u32(dev, "led-current", &param_val);
+	if (rc) {
+		dev_err(dev, "Couldn't get led current %x", rc);
+		goto exit;
+	}
+	led_current = param_val;
+
+	rc = device_property_read_u32(dev, "threshold", &param_val);
+	if (rc) {
+		dev_err(dev, "Couldn't get threshold %x", rc);
+		goto exit;
+	}
+	threshold = param_val;
+
+	rc = device_property_read_u32(dev, "count-exceed", &param_val);
+	if (rc) {
+		dev_err(dev, "couldn't get count exceed %x", rc);
+		goto exit;
+	}
+	count_exceed = param_val;
+
+	rc = i2c_smbus_read_byte_data(data->client, VCNL_PROD_REV);
+	if (rc < 0) {
+		dev_err(dev, "Error in prod rev reading out %x", rc);
+		goto exit;
+	}
+
+	if (rc == VCNL3020_PROD_ID) {
+		data->rev = rc & 0xff;
+		mutex_init(&data->vcnl3020_lock);
+	} else {
+		dev_err(dev, "Prod id is not related to vcnl3020");
+		rc = -ENODEV;
+		goto exit;
+	}
+
+	/* set proximity rate */
+	rc = i2c_smbus_write_byte_data(data->client, VCNL_PROXIMITY_RATE,
+				       proximity_rate);
+	if (rc < 0) {
+		dev_err(dev, "Error set proximity rate %x", rc);
+		goto exit;
+	}
+
+	/* set led current */
+	rc = i2c_smbus_write_byte_data(data->client, VCNL_LED_CURRENT,
+				       led_current);
+	if (rc < 0) {
+		dev_err(dev, "Error set led current %x", rc);
+		goto exit;
+	}
+
+	/* set high threshold value */
+	rc = i2c_smbus_write_byte_data(data->client, VCNL_PS_LO_THR_LO,
+				       (threshold) & 0xFF);
+	if (rc < 0) {
+		dev_err(dev, "Error set high threshold lreg %x", rc);
+		goto exit;
+	}
+
+	/* set low threshold value */
+	rc = i2c_smbus_write_byte_data(data->client, VCNL_PS_LO_THR_HI,
+				       (threshold >> 8) & 0xFF);
+	if (rc < 0) {
+		dev_err(dev, "Error set high threshold hreg %x", rc);
+		goto exit;
+	}
+
+	/* enable interrupt for exceeding low/high thresholds */
+	rc = i2c_smbus_write_byte_data(data->client, VCNL_PS_ICR,
+				       (count_exceed << ICR_COUNT_EXCEED) |
+				       ICR_THRES_EN);
+	if (rc < 0) {
+		dev_err(dev, "Error set interrupt control reg %x", rc);
+		goto exit;
+	}
+
+	rc = i2c_smbus_write_byte_data(data->client, VCNL_COMMAND,
+				       (VCNL_PS_EN | VCNL_PS_SELFTIMED_EN) &
+				       (VCNL_PS_OD ^ 0xff));
+	if (rc < 0) {
+		dev_err(dev, "Error set interrupt control reg %x", rc);
+		goto exit;
+	}
+
+	rc = i2c_smbus_read_byte_data(data->client, VCNL_COMMAND);
+	dev_err(dev, "command data %x", rc);
```

Comment:
Why is this a `dev_err()` ? Looks more like `dev_dbg()`.

Reply:
It was debug information, forgot to remove it.